sources/common/utils_test.go

```diff
@@ -137,3 +137,154 @@ func TestWorkerPool(t *testing.T) {
 	out, _ := RunParallelTasks(input, 5, f, false)
 	assert.Equal(t, len(input), len(out), fmt.Sprintln("jobs not processed"))
 }
+
+func TestPrepareColumns(t *testing.T) {
+	tc := []struct {
+		name           string
+		conv           *internal.Conv
+		tableId        string
+		srcCols        []string
+		expectedColIds []string
+	}{
+		{
+			name: "when source and spaner tables have same set of columns",
+			conv: &internal.Conv{
+				SpSchema: map[string]ddl.CreateTable{
+					"t1": {
+						Name:   "t1",
+						ColIds: []string{"c1", "c2"},
+						ColDefs: map[string]ddl.ColumnDef{
+							"c1": {Name: "a", Id: "c1", T: ddl.Type{Name: ddl.Int64}},
+							"c2": {Name: "b", Id: "c2", T: ddl.Type{Name: ddl.String, Len: 6}},
+						},
+						PrimaryKeys: []ddl.IndexKey{{ColId: "c1"}},
+					}},
+				SrcSchema: map[string]schema.Table{
+					"t1": {
+						Name:   "t1",
+						ColIds: []string{"c1", "c2"},
+						ColDefs: map[string]schema.Column{
+							"c1": {Name: "a", Id: "c1", Type: schema.Type{Name: "bigint", Mods: []int64{}}},
+							"c2": {Name: "b", Id: "c2", Type: schema.Type{Name: "varchar", Mods: []int64{6}}},
+						},
+						PrimaryKeys: []schema.Key{{ColId: "c1"}},
+					}},
+			},
+			tableId:        "t1",
+			srcCols:        []string{"a", "b"},
+			expectedColIds: []string{"c1", "c2"},
```

Comment:
We should perform deep validation of the conversion - e.g `column name`, `Len`, `datatype` instead of just the `id`.
Applicable to other tests in this file as well.

Reply:
PrepareColumns() and PrepareValues() return array of strings only, so didn't understand how to perform deep validation for column name , len, datatype etc?